### PR TITLE
implemented changes to ask the user before enabling bluetooth.

### LINKED ISSUE #293: Need a warning that we're turning on bluetooth
A couple people have commented that they were surprised that our app automatically turns on bluetooth.  Not necessarily in a "oh it shouldn't do that" way, but we probably should put up a small warning that "this app will enable your bluetooth radio."

### LINKED ISSUE #294: Bluetooth enabled when app starts
A couple people have commented about starting bluetooth when the app starts.  We don't actually need bluetooth until we attempt to scan for a guest or make ourselves findable by a host, so we should not enable bluetooth until then.

I made this an issue because I believe this isn't working as intended and is causing people confusion.

### LINKED ISSUE #298: Bluetooth enabled at random after SoundStream is installed
Some users have reported an issue where, after installing SoundStream, their bluetooth adapter will randomly enable itself.

### PR DESCRIPTION
**Pull request by theJenix**
removed auto enable of bluetooth from ConnectionService onCreate

added wrapper class to ask the user, enable bluetooth, and then run supplied
code

implemented use of wrapper on "Join" button (connectfragment), "Add users"
button, and "Switch networks" button (networkfragment)

Fixes #298 
Fixes #294 
Fixes #293

### CODE REVIEW

File: SoundStream/src/com/thelastcrusade/soundstream/components/WithBluetoothEnabled.java [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+                        @Override
+                        public void onClick(DialogInterface dialog, int which) {
+                            if (connectionService.enableNetwork()) {
+                                handler.post(runnable);
+                            } else {
+                                Log.wtf(TAG, "Error enabling bluetooth");
+                            }
+                        }
+                    })
+                   .setNegativeButton(android.R.string.cancel, new DialogInterface.OnClickListener() {
```
Comment (reidbaker) [inline SoundStream/src/com/thelastcrusade/soundstream/components/WithBluetoothEnabled.java:86]: I dont think we have to set an onclick listener if we dont want to do anything.

**Comment by reidbaker** [comment]
one small comment but I still think this is good to :punch:  in

File: SoundStream/src/com/thelastcrusade/soundstream/components/WithBluetoothEnabled.java (hunk 1 above)
Comment (theJenix) [inline SoundStream/src/com/thelastcrusade/soundstream/components/WithBluetoothEnabled.java:86]: I think if we don't, it wont display a cancel button (i'll verify this today).  If that's the case, do we prefer no cancel button, or a NOOP listener?

Comment (reidbaker) [inline SoundStream/src/com/thelastcrusade/soundstream/components/WithBluetoothEnabled.java:86]: I would prefer a noop listener to not having a cancel button

**Comment by dgreenhalgh** [comment]
Once Reid's comment is cleared up, let's :shipit:

**Comment by theJenix** [comment]
Hmm...it appears my latest commit cleared out comments about the cancel button...that's unfortunate.  Anyway, you must have a setNegativeButton call to get the cancel button, so I left in the no-op listener and added a comment explaining it.

**Comment by theJenix** [comment]
Are there any other comments on this?  Is this ready to go?

**Comment by dgreenhalgh** [comment]
If @reidbaker's happy, let's :shipit:

**Comment by taywrobel** [comment]
I had to look up android threading (Handler, really, are they that averse to normal conventions -.-), but after that I think this looks good.  Pulling in so we can get some testing in before the next patch release.

**Pull request merged**